addons/l10n_sa_edi/models/account_edi_format.py

```diff
@@ -476,3 +479,37 @@ def _get_move_applicability(self, move):
             'post': self._l10n_sa_post_zatca_edi,
             'edi_content': self._l10n_sa_get_invoice_content_edi,
         }
+
+    def _prepare_invoice_report(self, pdf_writer, edi_document):
+        """
+        Prepare invoice report to be printed.
+        :param pdf_writer: The pdf writer with the invoice pdf content loaded.
+        :param edi_document: The edi document to be added to the pdf file.
+        """
+        super()._prepare_invoice_report(pdf_writer, edi_document)
+        if self.code != 'sa_zatca' or edi_document.move_id.country_code != 'SA':
+            return
+
+        attachment = edi_document.attachment_id
+        if not attachment or not attachment.datas:
+            return  # nothing to embed
```

Comment:
Not necessary but logging a warning here would be a nice improvement
```suggestion
            _logger.warning(f"No attachment found for invoice {edi_document.move_id.name}")
            return